eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -201,14 +201,14 @@ class Router(val nodeParams: NodeParams, watcher: ActorRef, initialized: Option[
     case Event(n: NodeAnnouncement, d: Data) =>
       stay using Validation.handleNodeAnnouncement(d, nodeParams.db.network, Set(LocalGossip), n)
 
-    case Event(PeerRoutingMessage(peerConnection, remoteNodeId, n: NodeAnnouncement), d: Data) =>
-      stay using Validation.handleNodeAnnouncement(d, nodeParams.db.network, Set(RemoteGossip(peerConnection, remoteNodeId)), n)
+    case Event(PeerRoutingMessage(remoteNodeId, n: NodeAnnouncement), d: Data) =>
+      stay using Validation.handleNodeAnnouncement(d, nodeParams.db.network, Set(RemoteGossip(sender, remoteNodeId)), n)
```

Comment:
Don't you still have an issue with `RemoteGossip` containing an actor ref?

Reply:
Maybe in the future, but not now because we are only exchanging `PeerRoutingMessage` between `PeerConnection` and `Router` actors. The router then uses `GossipOrigin` internally, and with `Peer` but those are local.